code/modules/mob/living/simple_animal/simple_animal.dm

```diff
@@ -475,9 +477,22 @@
 			adjustBruteLoss(30)
 
 /mob/living/simple_animal/adjustBruteLoss(damage)
-	health = Clamp(health - damage, 0, maxHealth)
-	if(health < 1)
-		Die()
+	if(!ignored_damage_types[BRUTE])
+		health = Clamp(health - damage, 0, maxHealth)
+		if(health < 1)
+			Die()
+
+/mob/living/simple_animal/adjustFireLoss(damage)
+	if(!ignored_damage_types[BURN])
+		adjustBruteLoss(damage)

```

Comment:
Anything immune to brute will be immune to burn because this uses adjustBruteLoss().